### DISCUSSION TITLE
Changing size of toolbar icons in CSS

### DISCUSSION THREAD

**Discussion by advice2020** [geany/geany#4056] [Q&A]
Little confused.
When I try to adjust the toolbar icon sizes in `Preferences - Interface - Toolbar - Icon Size` all of the four options seem to use the same icon sizes in toolbar (unless maybe slight differences?).  Regardless I would like to try out larger icons and the "Large Icons" setting does not appear to generate what I am looking for.
I was told that this is not a bug and to try using CSS to achieve this.

I have read that GTK3 no longer supports icon sizes? not sure what this pertains to, is this icon sizes in general?

I set `Preferences - Interface - Toolbar - Icon Size` to `System default`
If I add the following to my `gtk.css` file it increases the size of the icons but because they are "scaled" with this approach both make icons appear blurry.
`#GeanyMainWindow toolbar toolbutton image {-gtk-icon-transform: scale(3.0);}`
`#GeanyMainWindow toolbar toolbutton image {-gtk-icon-transform: scale(2.0);}`

More I think about it I do not even know where these icons are coming from that Geany uses for toolbar?
I did some searching but did not find anything information.
I am currently using Papirus icon theme, are being generated from this set?

Does anyone know of a way to set to set larger size icons (larger pixel amount) / specific size icons (programs usually use 16x16 24x24 32x32 48x48) for Geany toolbar in `gtk.css` file?

Not very important but again just something that I wanted to try in Geany.
I am also just trying to learn a little bit more about how CSS can be used to alter program GUIs, in the past I learned about fonts/font sizes but have not learned that much about icons/icon sizes yet.

Thank You

**Comment by eht16** [comment]
> More I think about it I do not even know where these icons are coming from that Geany uses for toolbar? I did some searching but did not find anything information. 

All of the icons are read from the icon theme with a few exceptions: Save All, Close All and Build icons are provided by Geany as there are no suitable icons defined for icon themes, the icons can be found https://github.com/geany/geany/tree/master/icons/scalable.

> Does anyone know of a way to set to set larger size icons (larger pixel amount) / specific size icons (programs usually use 16x16 24x24 32x32 48x48) for Geany toolbar in `gtk.css` file?

Me not :).
Maybe it's better to ask this in some GTK or theming community as it is not related to Geany. AFAIK we do not anything special to the toolbar icons.

**Comment by advice2020** [comment]
@eht16 
Thank you for your response.
Okay, is it okay if I leave this post open a little while just in case maybe a different user has experience using both Geany and has GTK CSS knowlege that they can share about this.
I will also see if I can also find answers with what you have mentioned, if I come across anything I will also post it here before/after I close this post just in case other users are ever interested in this in the future.
Thank You
